### PR TITLE
Drop type specs in RecoB and RecoEgamma

### PR DESCRIPTION
**Pull request by NiharSaha**
#### PR description:
Drop type specs in Validation/RecoB and Validation/RecoEgamma python configuration files.

Central DQM migration campaign for drop type spces following 
https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1
and update the safer syntax for existing parameters like: (1) drop type specifications where the original parameter exists. (2) using "clone" instead of "deepcopy" (3) move all parameter inside the clone

#### PR validation:

Tested in CMSSW_12_3_X via runTheMatrix.py -l limited -i all –ibeos

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

File: Validation/RecoEgamma/python/tkConvValidator_cfi.py [hunk 1]
```diff
@@ -10,29 +10,29 @@
 trackAssociatorByHitsForConversionValidation.Cut_RecoToSim = 0.5
 
 from CommonTools.RecoAlgos.trackingParticleRefSelector_cfi import trackingParticleRefSelector as _trackingParticleRefSelector
-tpSelecForEfficiency = _trackingParticleRefSelector.clone()
-tpSelecForEfficiency.chargedOnly = True
+tpSelecForEfficiency = _trackingParticleRefSelector.clone(
+chargedOnly = True,
 # trackingParticleSelector.pdgId = cms.vint32()
```
Comment (perrotta) [inline Validation/RecoEgamma/python/tkConvValidator_cfi.py:15]: One could have already migrated also the commented out lines... but ok, lets defer it to when they'll get uncommented

**Comment by perrotta** [comment]
+1

File: Validation/RecoEgamma/python/tkConvValidator_cfi.py (hunk 1 above)
Comment (NiharSaha) [reply Validation/RecoEgamma/python/tkConvValidator_cfi.py:15]: @perrotta  I would change those commented lines in my next PR. Thanks!

**Pull request merged**